pkg/security/probe/sysctl/snapshot.go

```diff
@@ -247,6 +258,75 @@ func (s *Snapshot) snapshotKernelCmdline(ignoredBaseNames []string) error {
 	return nil
 }
 
+func (s *Snapshot) getKernelConfigPath() (string, error) {
+	kernelVersion, err := kernel.Release()
+	if err != nil {
+		return "", err
+	}
+	configPath := fmt.Sprintf(kernel.HostBoot("/config-%s"), strings.TrimSpace(string(kernelVersion)))
+	if _, err := os.Stat(configPath); err == nil {
+		return configPath, nil
+	}
+	procConfigGZ := kernel.HostProc("/config.gz")
+	if _, err := os.Stat(procConfigGZ); err == nil {
+		return procConfigGZ, nil
+	}
+	return "", fmt.Errorf("kernel config not found")
+}
+
+func (s *Snapshot) parseKernelConfig(r io.Reader, kernelCompilationFlags map[string]uint8) error {
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if strings.HasPrefix(line, "#") {
+			if strings.HasSuffix(line, "is not set") {
+				key := bytes.Fields([]byte(line))[1]
+				if _, ok := kernelCompilationFlags[string(key)]; ok {
+					s.KernelCompilationConfiguration[string(key)] = "not_set"
+				}
+			}
+			continue
+		}
+		parts := strings.SplitN(line, "=", 2)
```

Comment:
small nit, maybe for a follow up, but `strings.Cut` is much cleaner and doesn't allocate

Reply:
Done 👍🏻